### ISSUE TITLE
Support reverse mounts (VM data mounted on the host)

### ISSUE THREAD

**Issue by nammn** [canonical/multipass#1070]
Currently, mount support mounting a host dir on the vm and share them between the vm and host.

I use multipass to configure a local dev environment for my osx.

Is it possible to do it the other way around, mount a dir from the VM on the localhost (mac)?
Afaict it is not implemented, do you guys have some nice commands+options in mind which is designed for speed?

**Comment by Saviq** [comment]
Hi @nammn can you suggest a use case that isn't covered by mounting host data into the instance?

A couple reasons we decided to focus on that direction are: data security (less likely that you lose anything) and simplicity (mounting into host system is much more complex cross-platform).

But maybe we overlooked something :)

**Comment by nammn** [comment]
Hey @Saviq so I actually do most of my compiling and things in the vm (golang), which seems to need to use a linux file system for some parts.

OSX: Goland
VM: Go code, compiling

if I mount it vice versa the compiling is not working. RIght now I "just" sshfs to make it work.

Does it make sense? :)

**Comment by Saviq** [comment]
@nammn that sounds like a bug in our mount support then, we're actually using sshfs for the mounts into the instance.

Can you share what the error message is?

**Comment by nammn** [comment]
So it works if I mount the multipass folder on my osx (vm sshfs on osx).

It does not work if I mount osx path on vm (vm mounts osx)

My use case is, which does not work:
`go get -d -v github.com/juju/juju/...` (from osx)
`multipass mounts` vm  local gopath to linux gopath
`dep ensure` (from linux, which is getting dependencies and compiling some binaries)`

My use case i want, which works: <-- which seems more reasonable to me because of the compiling of binaries and things
`go get -d -v github.com/juju/juju/...` (from linux)
`sshfs vm on osx`
`dep ensure` from linux

error: 
```
something along 
waiting for lockfile dep
```

Could you share which sshfs option you guys use to optimize for local performance?

**Comment by Saviq** [comment]
Right, lock files… Not sure we'll be able to get those through sftp…

The sshfs process in the instance just has:
`sshfs -o slave -o nonempty -o transform_symlinks -o allow_other…`

But we have a custom sftp server that we've optimized, you can see the code [here](https://github.com/CanonicalLtd/multipass/tree/master/src/sshfs_mount).

**Comment by ricab** [comment]
It's been 6 years and we haven't seen a lot of interest in this. I am not sure if there is still an issue with lock files that cannot be worked around with `--type native` but, if so, I would prefer to track it in a separate issue.
